### DISCUSSION TITLE
How can I access the queries in getStaticProps

### DISCUSSION THREAD

**Discussion by Vince-vegas** [vercel/next.js#17659] [Help] [answered]
E.g 
You have a route that is **/movie?sorted=top_rated**

**1.**
is there any way that getStaticProps can access the **sorted** query for fetching the movie's sorted
or
**2.**
just implement **useRouter Hook** in the component then use the **useEffect Hook** and fetched the data

I've been thinking when I use the 2nd method the Next.js wont statically prerendered the data

**Comment by HaNdTriX** [comment, marked as answer]
1. No. This is by design. `getStaticProps` runs at **buildtime**, but the `query` object is created at **runtime**. 
2. Yes thats true. If you access the `query` at **runtime** the data is fetched at **runtime**

-----

You have multiple options:

## 1. Hybrid

- Decide for a good default e.g. `top_rated`
- Prerender data using `getStaticProps`
- Refetch on the client if `query.sorted !== top_rated`

## 2. Prerender all variations

- Use `sorted` has param e.g.: ` /movie/:sorted`
- Use `getStaticPaths` to prerender all options

## 3. Client only fetching

- Render a placeholder
- Fetch on the client

**Comment by Vince-vegas** [comment]
Thanks for this

**Comment by dpyzo0o** [comment]
@HaNdTriX Hi, is it possible to get the current route in getStaticProps? e.g, for pages `/foo/bar` how could I get the value `/foo/bar` inside the getStaticProps function?

**Reply by dpyzo0o** [reply]
My usecase is I would like to use a shared `getStaticProps` function for many pages, thus I need to know which page that is invoking the `getStaticProps` function.
